File: app/src/debug/java/com/azizbekian/spyur/injection/BuildTypeModule.java
package com.azizbekian.spyur.injection;

import android.os.SystemClock;

import com.facebook.stetho.okhttp3.StethoInterceptor;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;

/**
 * Created on July 29 2016.
 *
 * @author Andranik Azizbekian (andranik.azizbekyan@gmail.com)
 */
@Module
public class BuildTypeModule {

    @Provides @Singleton
    public OkHttpClient provideOkHttpClient() {
        OkHttpClient.Builder builder = new OkHttpClient.Builder();

        HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
        interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
        builder
                .addNetworkInterceptor(new StethoInterceptor())
                .addInterceptor(interceptor)
                .addInterceptor(chain -> {
                    // Simulating network delay
                    SystemClock.sleep(1500);
                    return chain.proceed(chain.request());
                });

        return builder.build();
    }

}

File: app/src/debug/java/com/incentive/yellowpages/injection/module/BuildTypeModule.kt
package com.incentive.yellowpages.injection.module

import com.facebook.stetho.okhttp3.StethoInterceptor
import dagger.Module
import dagger.Provides
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import javax.inject.Singleton

@Module
class BuildTypeModule {

    @Provides
    @Singleton
    fun provideOkHttpClient(): OkHttpClient.Builder {
        val builder = OkHttpClient.Builder()
        val logging = HttpLoggingInterceptor()
        logging.level = HttpLoggingInterceptor.Level.BASIC
        builder
                .addNetworkInterceptor(StethoInterceptor())
                .addInterceptor(logging)
                .addInterceptor { chain ->
                    // Simulating network delay
                    // SystemClock.sleep(3000)
                    chain.proceed(chain.request())
                }

        return builder
    }

}
